src/Immutable.php

```diff
@@ -25,7 +25,9 @@ public function __construct()
             }
 
             $this->_userDefinedProperties[$propertyName] = true;
-            $this->_defaultValues[$propertyName] = $property->getValue($this);
+            if ($property->isInitialized($this)) {
```

Comment:
Since `ReflectionProperty::isInitialized()` was added in `PHP 7.4` it should be as follows to not break usage in `PHP < 7.4`:

```suggestion
            if (\PHP_VERSION_ID < 70400 || $property->isInitialized($this)) {
```

Reply:
Thank you for this hint!